src/pages/Send/hooks/useSend/usePVMSend.ts

```diff
@@ -58,9 +87,82 @@ export const usePvmSend: SendAdapterPVM = ({
     setIsValidating(false);
   }
 
+  const getFeeState = useCallback(
+    (gasPrice?: bigint) => {
+      if (!gasPrice || !feeState) {
+        return feeState;
+      }
+
+      return {
+        ...feeState,
+        price: gasPrice,
+      };
+    },
+    [feeState]
+  );
+
+  const buildTransaction = useCallback(
+    async ({ address, amount, gasPrice, token }: PVMSendOptions) => {
+      const avax = provider.getAvaxID();
+      const amountBigInt = bigToBigInt(Big(amount), token.decimals);
+      const changeAddress = utils.parse(account.addressPVM)[2];
+      const { utxos } = await getMaxUtxoSet(
+        isLedgerWallet,
+        provider,
+        wallet,
+        network,
+        getFeeState(gasPrice)
+      );
+
+      return wallet.baseTX({
+        utxoSet: utxos,
+        chain: PCHAIN_ALIAS,
+        toAddress: address,
+        amountsPerAsset: {
+          [avax]: amountBigInt,
+        },
+        options: {
+          changeAddresses: [changeAddress],
+        },
+        feeState:
+          feeState && gasPrice ? { ...feeState, price: gasPrice } : feeState,
+      });
+    },
+    [
+      account.addressPVM,
+      provider,
+      wallet,
+      feeState,
+      isLedgerWallet,
+      network,
+      getFeeState,
+    ]
+  );
+
+  const parseTx = useCallback(
+    async ({ address, amount, gasPrice, token }: PVMSendOptions) => {
+      const unsignedTx = await buildTransaction({
+        address,
+        amount,
+        gasPrice,
+        token,
+      });
+
+      const parsedTx = await Avalanche.parseAvalancheTx(
+        unsignedTx,
+        provider,
+        account.addressPVM,
+        { feeTolerance: 100 }
```

Comment:
In this specific scenario, I was thinking about setting the tolerance based on the user's preferred fee rate.

For example, if the base fee is 10 nAvax and they choose to pay 15 nAvax, we'd set the tolerance here to `50`. At the same time going above 20 nAvax would result in errors, since `feeTolerance` must be in range of `[1, 100]`.